modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3050,6 +3049,12 @@ void GDScriptAnalyzer::reduce_subscript(GDScriptParser::SubscriptNode *p_subscri
 			if (base_type.is_variant() || !base_type.is_hard_type()) {
 				result_type.kind = GDScriptParser::DataType::VARIANT;
 				mark_node_unsafe(p_subscript);
+
+				reduce_identifier_from_base(p_subscript->attribute, &base_type);
+				GDScriptParser::DataType attr_type = p_subscript->attribute->get_datatype();
+				if (attr_type.is_set()) {
+					result_type = attr_type;
+				}
```

Comment:
This will set the type based on an inferred type, which is problematic. While it would be nice, inference is not yet completely trustworthy, so this might be dangerous since it will be used by the compiler to decide when to use typed instructions.

Reply:
Ah, maybe if this is also set as inferred then it might work.